File: Erik_Vincent/Assignment1/Assignment1/Model/QLS/AST/Style/Widget/CheckBox.cs
using Assignment1.Model.QL.AST;

namespace Assignment1.Model.QLS.AST.Style.Widget
{
    public class CheckBox : ASTNode, IWidget
    {
        public CheckBox(int lineNumber)
        {
            _lineNumber = lineNumber;
        }

        public void Accept(IStyleVisitor visitor) => visitor.Visit(this);
    }
}

File: GunOli/src/main/java/QLS/AST/Widgets/CheckBox.java
package QLS.AST.Widgets;

import QLS.Analysis.WidgetVisitorInterface;

public class CheckBox extends Widget {

    public CheckBox(int line) {
        super( WidgetType.CheckBox, line);
    }

    @Override
    public <T> T accept(WidgetVisitorInterface<T> visitor) {
        return visitor.visit(this);
    }
}
